### PR TITLE
Mark Framework interfaces as API part 9

### LINKED ISSUE #32036: Mark Framework interfaces as API part 9
The following interfaces should be marked as API in 2.4-develop branch

1. lib/internal/Magento/Framework/View/File/FileList/CollateInterface.php
1. lib/internal/Magento/Framework/View/Helper/SecureHtmlRender/SecurityProcessorInterface.php
1. lib/internal/Magento/Framework/View/Layout/Argument/UpdaterInterface.php
1. lib/internal/Magento/Framework/View/Layout/Condition/VisibilityConditionInterface.php
1. lib/internal/Magento/Framework/View/Layout/GeneratorInterface.php
1. lib/internal/Magento/Framework/View/Layout/LayoutCacheKeyInterface.php
1. lib/internal/Magento/Framework/View/Layout/ProcessorInterface.php
1. lib/internal/Magento/Framework/View/Layout/ReaderInterface.php
1. lib/internal/Magento/Framework/View/Model/PageLayout/Config/BuilderInterface.php
1. lib/internal/Magento/Framework/View/Page/Config/RendererInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/AttributeInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/CdataInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/CommentInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/Directive/DirectiveInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/Element/ElementInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/TextInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/ResultInterface.php
1. lib/internal/Magento/Framework/View/TemplateEngineInterface.php
1. lib/internal/Magento/Framework/View/Xsd/Media/TypeDataExtractorInterface.php
1. lib/internal/Magento/Framework/Webapi/CustomAttribute/PreprocessorInterface.php
1. lib/internal/Magento/Framework/Webapi/CustomAttribute/ServiceTypeListInterface.php
1. lib/internal/Magento/Framework/Webapi/Rest/Request/DeserializerInterface.php

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Marked following interfaces as API  + Added strict_types

lib/internal/Magento/Framework/View/File/FileList/CollateInterface.php
lib/internal/Magento/Framework/View/Helper/SecureHtmlRender/SecurityProcessorInterface.php
lib/internal/Magento/Framework/View/Layout/Argument/UpdaterInterface.php
lib/internal/Magento/Framework/View/Layout/Condition/VisibilityConditionInterface.php
lib/internal/Magento/Framework/View/Layout/GeneratorInterface.php
lib/internal/Magento/Framework/View/Layout/LayoutCacheKeyInterface.php
lib/internal/Magento/Framework/View/Layout/ProcessorInterface.php
lib/internal/Magento/Framework/View/Layout/ReaderInterface.php
lib/internal/Magento/Framework/View/Model/PageLayout/Config/BuilderInterface.php
lib/internal/Magento/Framework/View/Page/Config/RendererInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/AttributeInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/CdataInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/CommentInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/Directive/DirectiveInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/Element/ElementInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/Compiler/TextInterface.php
lib/internal/Magento/Framework/View/TemplateEngine/Xhtml/ResultInterface.php
lib/internal/Magento/Framework/View/TemplateEngineInterface.php
lib/internal/Magento/Framework/View/Xsd/Media/TypeDataExtractorInterface.php
lib/internal/Magento/Framework/Webapi/CustomAttribute/PreprocessorInterface.php
lib/internal/Magento/Framework/Webapi/CustomAttribute/ServiceTypeListInterface.php
lib/internal/Magento/Framework/Webapi/Rest/Request/DeserializerInterface.php

marked references api/class related
lib/internal/Magento/Framework/View/Helper/SecureHtmlRender/TagData.php
lib/internal/Magento/Framework/View/Helper/SecureHtmlRender/EventHandlerData.php
lib/internal/Magento/Framework/View/PageLayout/Config.php

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32036

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Rebased 2.5-develop branch

**Comment by eduard13** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Public type 'Magento\Framework\View\Model\PageLayout\Config\BuilderInterface' references following non-public types:
\Magento\Framework\View\PageLayout\Config
Seems add `api ` annotation got issue fail static tests
Magento\Test\Integrity\PublicCodeTest->testAllPHPClassesReferencedFromPublicClassesArePublic()
CC: @sivaschenko

**Comment by eduard13** [comment]
@mrtuvn, looks like we'll have to mark them as api as well.

**Comment by mrtuvn** [comment]
Fixed static

**Comment by mrtuvn** [comment]
Fixed static
@magento run all tests

File: lib/internal/Magento/Framework/View/Page/Config/RendererInterface.php [hunk 1]
```diff
@@ -3,21 +3,20 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
-
 namespace Magento\Framework\View\Page\Config;
 
 /**
  * Interface RendererInterface
  *
  * @package Magento\Framework\View\Page\Config
```
Comment (eduard13) [inline lib/internal/Magento/Framework/View/Page/Config/RendererInterface.php]: Please let remove the `package` annotation, as not being needed.

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @mrtuvn, please review the below minor comment, and can you please declare the strict types for all the classes as well?

Thank you.

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE

**Comment by eduard13** [comment]
@magento run Functional Tests CE

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8813](https://jira.corp.magento.com/browse/ENGCOM-8813) has been created to process this Pull Request

**Comment by mrtuvn** [comment]
Re-targeted PR branch to 2.4-develop as internal discussed with @gabrieldagama

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Integration Tests

**Comment by gabrieldagama** [comment]
@magento import code to https://github.com/magento-engcom/magento2ce

**Comment by magento-engcom-team** [comment]
@gabrieldagama the branch with code successfully imported into`magento-engcom/magento2ce` repository. Branch name: `imported-magento-magento2-32151`.

**Pull request merged**